lib/transaction.js

```diff
@@ -85,7 +85,7 @@ Transaction.begin = function(connector, options, cb) {
       return cb(err);
     }
     var tx = connection;
-    if (!(connection instanceof Transaction)) {
+    if (!(connection instanceof Transaction) && !(typeof connection.txId === 'string')) {
```

Comment:
I would like to see test cases around this code, though. You should be able to pass it a compliant object, as well as a non-compliant object (for the unhappy path).